src/main/java/MyCallable.java

```diff
@@ -0,0 +1,44 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+public class MyCallable implements Callable<Integer> {
+    private static final int LIMIT = 1000;
+    private final List<Integer> numbers;
+
+    public MyCallable(List<Integer> numbers) {
+        this.numbers = numbers;
+    }
+
+    @Override
+    public Integer call() throws Exception {
+        return numbers.stream().mapToInt(Integer::valueOf).sum();
+    }
+
+    public static int getSum(List<Integer> list, int threads) {
+        if (threads <= 0 || threads > LIMIT) {
+            throw new RuntimeException("Incorrect number of threads");
+        }
+        ExecutorService executor = Executors.newFixedThreadPool(threads);
```

Comment:
separate this logic into 2 different classes. First is callable impl that performs some atomic task. Second class will manage ExecutorService and use Callable impls inside